### PR TITLE
[WOR-85] Enable cloning for Azure workspaces, use `cloudPlatform` from workspace info to determine Azure vs. GCP workspaces.

### PR DESCRIPTION
**Pull request by cahrens**
This PR enables the Clone menu item for Azure workspaces-- the enable/disable state of workspace menu items is now identical for GCP and Azure workspaces.

As part of this change, I have standardized how we determine a GCP vs. an Azure workspace (this was actually necessary for Clone to behave correctly when launched from the workspace list, where we do not have `azureContext` available).

I also fixed a flashing bug that I noticed during the workspace list loading.

Things to test:

1. Clone of an Azure workspace from both the workspace dashboard and the list view (the error you get with the message "WOR-625" is expected). Only Azure billing projects should be presented.
2. No flashing of persistent disk message while loading workspaces list.
3. Correct tabs displayed for Azure vs. GCP workspaces (no change in behavior).
4. Correct cloud information displayed on workspace dashboard (no change in behavior).
5. Share workspace modal behaves appropriately for both Azure and Google workspaces, when triggered from either the list or dashboard view (no change in behavior).
6. Duplication of analysis.

### CODE REVIEW

File: src/libs/workspace-utils.ts
Comment (cahrens) [inline src/libs/workspace-utils.ts:7]: This was unused. Is there any reason to keep it?

File: src/libs/workspace-utils.ts
Comment (cahrens) [inline src/libs/workspace-utils.ts:29]: Either check is valid, but changing to verifying `cloudPlatform` for consistency with `isAzureWorkspace`.

File: src/pages/workspaces/List.js
Comment (cahrens) [inline src/pages/workspaces/List.js:106]: This avoids the message flashing while we are in the state of `workspaces` being an empty list (which is truthy) and the `disks` variable being populated.

**Comment by cahrens** [comment]
Integration test failure (find-workflow) is unrelated. Rerunning.

**Review by marctalbott — APPROVED** [review]
Tested it out, everything looked good

File: src/components/NewWorkspaceModal.js
Comment (ehigham) [inline src/components/NewWorkspaceModal.js]: perhaps test that it's a google billing project directly - ie call a function called `isGcpBillingProject` (that could be implemented this way)

File: src/pages/workspaces/List.js
Comment (ehigham) [reply src/pages/workspaces/List.js:106]: TIL - empty list is truthy. because of course it is. Thanks for fixing.

**Review by ehigham — APPROVED** [review]

File: src/components/NewWorkspaceModal.js
Comment (cahrens) [reply src/components/NewWorkspaceModal.js]: Sure, done.

**Comment by cahrens** [comment]
This PR is in draft until https://github.com/broadinstitute/rawls/pull/2162 is merged.

File: src/libs/workspace-utils.ts
Comment (cahrens) [reply src/libs/workspace-utils.ts:29]: It actually turns out that in some Analysis usages, they only pull out and pass around select things from the Workspace details. Therefore switching this to `cloudPlatform` would require changing some of their code to include `cloudPlatform` (else renaming and copying of Analyses breaks). To minimize impact, switching back to checking `googleProject`.

Comment (cahrens) [reply src/libs/workspace-utils.ts:29]: Actually, looking at the [unit tests](https://github.com/DataBiosphere/terra-ui/blob/9170b4cbf86ee32723290dcd55cc9b078284a78a/src/pages/workspaces/workspace/analysis/modals/AnalysisDuplicator.test.ts#L81) and the [TS interface](https://github.com/DataBiosphere/terra-ui/pull/3607/files#diff-f4ab958f7e080693ca641686dc6363514fd16b399acf1fd682a65f4be007a06eR11), `cloudPlatform` is expected to be in `BaseWorkspaceInfo`, so I have changed the calling locations to include it. FYI @jdcanas and @n8landolt 

I am surprised that typescript wasn't complaining about WorkspaceInfo missing the `cloudPlatform`. @nawatts, do you know why?

Comment (nawatts) [reply src/libs/workspace-utils.ts:29]: It looks like this PR adds `cloudPlatform` in two JS files (Analyses.js and AnalysisLauncher.js). Those files would not be checked by TypeScript.

Comment (cahrens) [reply src/libs/workspace-utils.ts:29]: Oh right. I was thinking that it would be caught because `isGoogleWorkspaceInfo` is called from `AnalysisDuplicator`, which is in TypeScript, but I guess that the whole chain has to be TypeScript for the input argument to be analyzed properly.

Comment (nawatts) [reply src/libs/workspace-utils.ts:29]: Yeah, `AnalysisDuplicator`'s `workspaceInfo` prop is typed as `WorkspaceInfo`.
https://github.com/DataBiosphere/terra-ui/blob/0e178702932416959b76a04ab790270b48dbd111/src/pages/workspaces/workspace/analysis/modals/AnalysisDuplicator.ts#L35

So TS will validate that prop can be passed into `isGoogleWorkspaceInfo`.
https://github.com/DataBiosphere/terra-ui/blob/0e178702932416959b76a04ab790270b48dbd111/src/pages/workspaces/workspace/analysis/modals/AnalysisDuplicator.ts#L63

But it can't check what's passed to `AnalysisDuplicator` in `Analyses` because Analyses.js is JS not TS.
https://github.com/DataBiosphere/terra-ui/blob/0e178702932416959b76a04ab790270b48dbd111/src/pages/workspaces/workspace/analysis/Analyses.js#L474

**Comment by cahrens** [comment]
@jdcanas or @n8landolt this PR is still draft while waiting for the cloning back end changes to land. But it would be good for one of you to take a look at the Analysis-related changes. TIA

**Comment by cahrens** [comment]
@jdcanas or @n8landolt this PR may be merged soon. Can one of you take a look wrt to the Analysis changes?

**Review by n8landolt — APPROVED** [review]

**Comment by cahrens** [comment]
Test failure is known run-workflow failure. Merging to unblock team work.

**Pull request merged**
